azure-pipelines/pipelines.yml

```diff
@@ -19,6 +19,8 @@ jobs:
       failOnStderr: true
     - bash: build.amd64.debug/vcpkg-test
       displayName: 'Run vcpkg tests'
+      env:
+        VCPKG_ROOT: UNIT_TESTS_SHOULD_NOT_USE_VCPKG_ROOT

```

Comment:
I would prefer to do this in https://github.com/microsoft/vcpkg-tool/blob/abe9562a24a2e8fff13af310ef7d6164202350fa/src/vcpkg-test/catch.cpp#L9 rather than changing all our running infrastructure because that still means there's external setup needed to run the unit test binary.